include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -19,7 +19,7 @@
 
 #include "function.h"
 #include "binary_search_extension_defs.h"
-#include "oneapi/dpl/iterator"
+#include "../iterator"
```

Comment:
`iterator` is a high-level header. Could you please use an internal header `iterator_impl.h`? Will it work?